pkg/kubernetes/cluster.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+)
+
+// ClusterDomain returns the k8s cluster domain extracted from the
+// /etc/resolv.conf.
+func ClusterDomain(resolvConf io.Reader) (string, error) {
```

Comment:
Wouldn't it make more sense to pass in the result of `ioutil.ReadFile` instead? (I assume you're passing in the file here for ease of testing.)

Reply:
Also, would it make sense to use something like [github.com/containers/libpod/pkg/resolvconf](https://pkg.go.dev/github.com/containers/libpod/pkg/resolvconf) to do the parsing? Mostly because it (being part of podman and derived from docker) probably has seen enough `/etc/resolv.conf` files in the wild that it would be able to handle any unexpectedly weird configurations.

Reply:
> Wouldn't it make more sense to pass in the result of ioutil.ReadFile instead? (I assume you're passing in the file here for ease of testing.)

I'm not passing it for easing the testing, but it helps, yes. I don't see any advantage of doing the reading ahead of time. So, I'll keep it as is.

> Also, would it make sense to use something like github.com/containers/libpod/pkg/resolvconf to do the parsing?

Makes sense! I'll bring it here.